sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs

```diff
@@ -58,7 +58,7 @@ public RecordMatcher(bool compareBodies = true)
             "x-ms-correlation-request-id"
         };
 
-        public virtual RecordEntry FindMatch(RecordEntry request, IList<RecordEntry> entries)
+        public virtual RecordEntry FindMatch(RecordEntry request, IList<RecordEntry> entries, bool skipRequestBody)
```

Comment:
Why does FindMatch has to flow up to here?

Can we just clean request.Body before passing it in like we do in RecordingTransport before storing?